terraform/ec2-test.tf

```diff
@@ -0,0 +1,101 @@
+module "test_vpc" {
+  source = "github.com/terraform-aws-modules/terraform-aws-vpc"
+
+#   name = "${var.vpc_name}"
+
+#   cidr = "${var.vpc_cidr}"
+#   public_subnets  = ["${var.app_public_subnet_cidrs}"]
+#   private_subnets = ["${var.app_private_subnet_cidrs}"]
+#   database_subnets = ["${var.database_subnet_cidrs}"]
+
+#   enable_nat_gateway = "${var.enable_nat_gateway}"
+#   enable_dns_hostnames = "${var.enable_dns_hostnames}"
+#   enable_dns_support = "${var.enable_dns_support}"
+
+#   azs = ["${var.aws_az1}", "${var.aws_az2}"]
+
+#   create_database_subnet_group = "${var.create_database_subnet_group}"
+
+name = "ec2-test"
+cidr = "10.0.0.0/16"
+public_subnets = ["10.0.1.0/24"]
+enable_nat_gateway = "false"
+enable_dns_hostnames = "true"
+enable_dns_support = "true"
+azs = ["us-east-1c", "us-east-1d"]
+}
+
+data "template_file" "user_data_template" {
+  depends_on = ["aws_kinesis_firehose_delivery_stream.extended_s3_stream"]
+  template = "${file("${path.module}/files/ec2-test.tpl")}"
+
+  vars {
+    deliverystream = "${var.es_kinesis_delivery_stream}"
+  }
+}
+
+# resource "local_file" "user_data_rendered" {
+#     content = "${data.template_file.user_data_template.rendered}"
+#     filename = "${path.module}/files/ec2-test.sh"
+# }
```

Comment:
Might be a bit cleaner to do both the Kinesis agent setup and testing through Ansible. We'll want an Ansible role for users to install the Kinesis agent anyway, right?

Reply:
Actually, I think there's already an ansible role to install it, I just chose to go with the user-data route since I had to do some other manipulations on instance creation anyway. Thought maybe I'd keep the code simple since that was the only reason to use ansible at all.